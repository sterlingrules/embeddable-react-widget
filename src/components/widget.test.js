import React from 'react';
import { mount } from 'enzyme';
import Widget from './widget';
import { waitForSelection } from '../test-helpers';

describe('<Widget />', () => {
  test('open/close', async () => {
    const widgetDom = mount(<Widget />);
    expect(widgetDom).toMatchSnapshot();

    {
      const dockAnchorEl = widgetDom.find('button.dock');
      expect(dockAnchorEl).toHaveLength(1);
      // open widget
      dockAnchorEl.simulate('click');
    }

    expect(widgetDom).toMatchSnapshot();

    // dock does not exist anymore
    expect(widgetDom.find('a.dock')).toHaveLength(0);

    const closeAnchorEl = await waitForSelection(widgetDom, 'button.widget-header-icon');

    expect(closeAnchorEl).toHaveLength(1);
    // close widget
    closeAnchorEl.simulate('click');

    {
      const dockAnchorEl = await waitForSelection(widgetDom, 'button.dock');
      expect(dockAnchorEl).toHaveLength(1);
    }

    expect(widgetDom).toMatchSnapshot();
  });
});
