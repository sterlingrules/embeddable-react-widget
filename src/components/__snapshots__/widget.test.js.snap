// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Widget /> open/close 1`] = `
<Widget
  bodyText="Body"
  footerText="Footer"
  headerText="Header"
>
  <div
    className="docked-widget"
  >
    <Transition
      appear={false}
      enter={true}
      exit={true}
      in={false}
      mountOnEnter={false}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      timeout={250}
      unmountOnExit={false}
    >
      <div
        className="widget widget-exited"
      >
        <div
          className="widget-header"
        >
          <div
            className="widget-header-title"
          >
            Header
          </div>
          <button
            className="widget-header-icon"
            onClick={[Function]}
            onKeyPress={[Function]}
            type="button"
          >
            X
          </button>
        </div>
        <div
          className="widget-body"
        >
          Body
        </div>
        <div
          className="widget-footer"
        >
          Footer
        </div>
      </div>
    </Transition>
    <button
      className="dock"
      onClick={[Function]}
      onKeyPress={[Function]}
      type="button"
    >
      ^ OPEN ^
    </button>
  </div>
</Widget>
`;

exports[`<Widget /> open/close 2`] = `
<Widget
  bodyText="Body"
  footerText="Footer"
  headerText="Header"
>
  <div
    className="docked-widget"
  >
    <Transition
      appear={false}
      enter={true}
      exit={true}
      in={true}
      mountOnEnter={false}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      timeout={250}
      unmountOnExit={false}
    >
      <div
        className="widget widget-entering"
      >
        <div
          className="widget-header"
        >
          <div
            className="widget-header-title"
          >
            Header
          </div>
          <button
            className="widget-header-icon"
            onClick={[Function]}
            onKeyPress={[Function]}
            type="button"
          >
            X
          </button>
        </div>
        <div
          className="widget-body"
        >
          Body
        </div>
        <div
          className="widget-footer"
        >
          Footer
        </div>
      </div>
    </Transition>
  </div>
</Widget>
`;

exports[`<Widget /> open/close 3`] = `
<Widget
  bodyText="Body"
  footerText="Footer"
  headerText="Header"
>
  <div
    className="docked-widget"
  >
    <Transition
      appear={false}
      enter={true}
      exit={true}
      in={false}
      mountOnEnter={false}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      timeout={250}
      unmountOnExit={false}
    >
      <div
        className="widget widget-exited"
      >
        <div
          className="widget-header"
        >
          <div
            className="widget-header-title"
          >
            Header
          </div>
          <button
            className="widget-header-icon"
            onClick={[Function]}
            onKeyPress={[Function]}
            type="button"
          >
            X
          </button>
        </div>
        <div
          className="widget-body"
        >
          Body
        </div>
        <div
          className="widget-footer"
        >
          Footer
        </div>
      </div>
    </Transition>
    <button
      className="dock"
      onClick={[Function]}
      onKeyPress={[Function]}
      type="button"
    >
      ^ OPEN ^
    </button>
  </div>
</Widget>
`;
